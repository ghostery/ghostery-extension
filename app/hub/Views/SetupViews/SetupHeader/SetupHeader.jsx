/**
 * Setup Header Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';

/**
 * A Functional React component for rendering the Setup View
 * @return {JSX} JSX for rendering the Setup Header View of the Hub app
 * @memberof HubComponents
 */
const SetupHeader = props => (
	<div className="row align-center">
		<div className="columns small-12 large-10">
			<div className="SetupHeader flex-container align-center-middle">
				<img src={props.titleImage} />
				<div className="SetupHeader__title">
					<h3 dangerouslySetInnerHTML={{ __html: props.title }} />
				</div>
			</div>
		</div>
	</div>
);

// PropTypes ensure we pass required props of the correct type
SetupHeader.propTypes = {
	title: PropTypes.string.isRequired,
	titleImage: PropTypes.string.isRequired,
};

export default SetupHeader;
