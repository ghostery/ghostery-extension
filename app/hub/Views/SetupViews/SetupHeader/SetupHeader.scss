/**
 * Setup Blocking View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Setup Header
.SetupHeader {
	margin: 80px 60px 40px;
}
.SetupHeader img {
	max-width: 180px;
}
.SetupHeader__title h3 {
	font-size: 24px;
	line-height: 1.75;
	color: #4a4a4a;
	margin: 0 25px;
}
