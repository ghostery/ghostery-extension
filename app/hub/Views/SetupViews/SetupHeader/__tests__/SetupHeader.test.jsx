/**
 * Setup Header Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SetupHeader from '../SetupHeader';

describe('app/hub/Views/SetupViews/SetupHeader component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('setup header is rendered correctly', () => {
			const initialState = {
				title: 'Test Title',
				titleImage: 'test-title-image',
			};

			const component = renderer.create(
				<SetupHeader {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				title: 'Test Title',
				titleImage: 'test-title-image',
			};

			const component = shallow(<SetupHeader {...initialState} />);
			expect(component.find('.SetupHeader').length).toBe(1);
			expect(component.find('.SetupHeader img').length).toBe(1);
			expect(component.find('.SetupHeader__title h3').length).toBe(1);
		});
	});
});
