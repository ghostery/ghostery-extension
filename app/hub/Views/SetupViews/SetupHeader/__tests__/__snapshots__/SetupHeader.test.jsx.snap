// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SetupViews/SetupHeader component Snapshot tests with react-test-renderer setup header is rendered correctly 1`] = `
<div
  className="row align-center"
>
  <div
    className="columns small-12 large-10"
  >
    <div
      className="SetupHeader flex-container align-center-middle"
    >
      <img
        src="test-title-image"
      />
      <div
        className="SetupHeader__title"
      >
        <h3
          dangerouslySetInnerHTML={
            Object {
              "__html": "Test Title",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
